compression/compression.go

```diff
@@ -0,0 +1,163 @@
+package compression
+
+import (
+	"bytes"
+	"compress/bzip2"
+	"compress/gzip"
+	"fmt"
+	"io"
+	"os/exec"
+
+	"github.com/Sirupsen/logrus"
+	"github.com/docker/containerd/ioutils"
+	"github.com/docker/containerd/pools"
```

Comment:
We wanted to avoid pulling in or migrating this package. Same goes for `ioutils`, their functionality is limited and save very little over just using `bufio` and `sync` directly.